src/processor/operator/persistent/rel_batch_insert.cpp

```diff
@@ -225,5 +226,15 @@ void RelBatchInsert::finalizeInternal(ExecutionContext* context) {
     partitionerSharedState->partitioningBuffers[relInfo->partitioningIdx].reset();
 }
 
+void RelBatchInsert::updateProgress(ExecutionContext* context) {
+    std::lock_guard<std::mutex> lock(progressSharedState->mtx);
+    progressSharedState->partitionsDone = partitionerSharedState->nextPartitionIdx;
+    progressSharedState->partitionsTotal =
+        partitionerSharedState->numPartitions[partitionerSharedState->numPartitions.size() - 1];
```

Comment:
This is not the correct way to fetch `numPartitions`, you should make use of `partitioningIdx` inside `RelBatchInsertInfo` to grab the correct num partitions for the current directed rel table data being ingested.